addons/account_update_tax_tags/tests/test_account_update_tax_tags_wizard.py

```diff
@@ -0,0 +1,325 @@
+        """ Test in case there are multiple taxes set on the invoice line. """
+        tax_2 = self._create_tax('update_test_tax_2', tag_names=[
+            f'{self.tag_names[0]}_2',
+            f'{self.tag_names[1]}_2',
+            f'{self.tag_names[2]}_2',
+            f'{self.tag_names[3]}_2'
+        ])
+        move = self._create_move(tax_ids=self.tax_1 + tax_2)
+        self._change_tax_tag(self.tax_1, 'invoice_tax_tag_changed', invoice=True, base=False)
+        self._change_tax_tag(tax_2, 'invoice_tax_tag_changed_2', invoice=True, base=False)
+        self.wizard.update_amls_tax_tags()
+
+        invoice_line, tax_line, counterpart_line = self._get_amls_by_type(move)
+        self.assertEqual(invoice_line.tax_tag_ids.mapped('name'), ['update_test_invoice_base_tag_id', 'update_test_invoice_base_tag_id_2'])  # should not change
+        self.assertEqual(tax_line.tax_tag_ids.mapped('name'), ['invoice_tax_tag_changed', 'invoice_tax_tag_changed_2'])  # should change
+        self.assertFalse(counterpart_line.tax_tag_ids)  # should not change
+
+    def test_update_multi_company(self):
+        """ Tests that only the company that is selected when opening the wizard will have its amls updated. """
+        move_1 = self._create_move(tax_ids=self.tax_1)
+        self._change_tax_tag(self.tax_1, 'invoice_tax_tag_changed_for_company_1', invoice=True, base=False)
+        be_country_id = self.env.ref('base.be').id
+        company_2 = self.company_data_2['company']
+        company_2.write({'country_id': be_country_id, 'account_tax_fiscal_country_id': be_country_id})
+        self.env.user.company_id = company_2
+        tax_2 = self._create_tax(
+            'update_test_tax_2',
+            tag_names=[None, 'update_test_invoice_tax_tag_company_2', None, None]
+        )
+        move_2 = self._create_move(tax_ids=tax_2)
+        self._change_tax_tag(tax_2, 'invoice_tax_tag_changed_for_company_2', invoice=True, base=False)
+        self.wizard.update_amls_tax_tags()
+
+        tax_line_1 = move_1.line_ids.filtered(lambda aml: aml.tax_line_id)
+        tax_line_2 = move_2.line_ids.filtered(lambda aml: aml.tax_line_id)
+        # Only the first move_id should be updated since it belongs to first company which was initialized in the wizard
+        self.assertEqual(tax_line_1.tax_tag_ids.name, 'invoice_tax_tag_changed_for_company_1')
+        self.assertEqual(tax_line_2.tax_tag_ids.name, 'update_test_invoice_tax_tag_company_2')
+
+    def test_update_all_move_type(self):
+        """ Tests that all move type are correctly updated with the corresponding tags. """
+        move_types = [
+            # 'entry' tested in test_update_move_type_entry since it's more complex
+            'out_invoice',
+            'in_invoice',
+            'out_refund',
+            'in_refund',
+            'in_receipt',
+            'out_receipt',
+        ]
+        for move_type in move_types:
+            for line_type in ['base', 'tax']:
```

Comment:
In case you would prefer to remove nested loop
```suggestion       
        for move_type, line_type in itertools.product(move_types, ['base', 'tax']):
```